lib/ClangImporter/ImporterImpl.h

```diff
@@ -1891,43 +1891,20 @@ static inline Type applyToFunctionType(
   return type;
 }
 
-inline Optional<const clang::EnumDecl *> findAnonymousEnumForTypedef(
-    const ASTContext &ctx,
-    const clang::TypedefType *typedefType) {
-  auto *typedefDecl = typedefType->getDecl();
-  auto *lookupTable = ctx.getClangModuleLoader()->findLookupTable(typedefDecl->getOwningModule());
-
-  auto foundDecls = lookupTable->lookup(
-      SerializedSwiftName(typedefDecl->getName()), EffectiveClangContext());
-
-  auto found = llvm::find_if(foundDecls, [](SwiftLookupTable::SingleEntry decl) {
-    return decl.is<clang::NamedDecl *>() &&
-        isa<clang::EnumDecl>(decl.get<clang::NamedDecl *>());
-  });
-
-  if (found != foundDecls.end())
-    return cast<clang::EnumDecl>(found->get<clang::NamedDecl *>());
-
-  // If a swift_private attribute has been attached to the enum, its name will
-  // be prefixed with two underscores
-  llvm::SmallString<32> swiftPrivateName;
-  swiftPrivateName += "__";
-  swiftPrivateName += typedefDecl->getName();
-  foundDecls = lookupTable->lookup(
-      SerializedSwiftName(ctx.getIdentifier(swiftPrivateName)),
-      EffectiveClangContext());
-
-  auto swiftPrivateFound =
-      llvm::find_if(foundDecls, [](SwiftLookupTable::SingleEntry decl) {
-        return decl.is<clang::NamedDecl *>() &&
-               isa<clang::EnumDecl>(decl.get<clang::NamedDecl *>()) &&
-               decl.get<clang::NamedDecl *>()
-                   ->hasAttr<clang::SwiftPrivateAttr>();
-      });
-
-  if (swiftPrivateFound != foundDecls.end())
-    return cast<clang::EnumDecl>(swiftPrivateFound->get<clang::NamedDecl *>());
-
+inline Optional<const clang::EnumDecl *>
+findAnonymousEnumForTypedef(const ASTContext &ctx,
+                            const clang::TypedefType *typedefType) {
+
+  assert(typedefType && "Must provide non-null typedef");
+  const clang::TypedefNameDecl *typedefDecl = typedefType->getDecl();
+  const clang::DeclContext *declContext = typedefDecl->getDeclContext();
+  for (const clang::Decl *siblingDecl : declContext->decls()) {
```

Comment:
I don't think we can rely on this. It seems rather fragile to assume that these decls will always be next to each other. Also, can we guarantee that the child decls here will have a stable order?

Reply:
Well in the case of NS_OPTION, they are next to one another, and IIUC, this is only designed to handle that case. I don't see why the stability of the order matters.